lib/shoulda/matchers/active_record/association_matcher.rb

```diff
@@ -1472,6 +1579,16 @@ def foreign_key_reflection
           end
         end
 
+        def foreign_type
+          type = if [:has_one, :has_many].include?(macro)
```

Comment:
To simplify this expression and remove the temporary assignment, what do you think about moving the `.to_s` to `has_column?` at line 1509?